apps/rush-lib/src/scripts/install-run-common.ts

```diff
@@ -0,0 +1,299 @@
+ *
+ * Does not support "." or ".." path segments.
+ * Assumes the baseFolder exists.
+ */
+function ensureAndResolveFolder(baseFolder: string, ...pathSegments: string[]): string {
+  let resolvedDirectory: string = baseFolder;
+  try {
+    for (let pathSegment of pathSegments) {
+      pathSegment = pathSegment.replace(/[\\\/]/g, '_');
+      resolvedDirectory = path.resolve(resolvedDirectory, pathSegment);
+      if (!fs.existsSync(resolvedDirectory)) {
+        fs.mkdirSync(resolvedDirectory);
+      }
+    }
+  } catch (e) {
+    throw new Error(`Error building local installation directory (${path.resolve(baseFolder, ...pathSegments)}): ${e}`);
+  }
+
+  return resolvedDirectory;
+}
+
+function copyNpmrcIfItExists(rushCommonFolder: string, packageInstallFolder: string): void {
+  const npmrcPath: string = path.join(rushCommonFolder, 'config', 'rush', '.npmrc');
+  const packageInstallNpmrcPath: string = path.join(packageInstallFolder, '.npmrc');
+  if (fs.existsSync(npmrcPath)) {
+    try {
+      let npmrcFileLines: string[] = fs.readFileSync(npmrcPath).toString().split('\n');
+      npmrcFileLines = npmrcFileLines.map((line) => (line || '').trim());
+      const resultLines: string[] = [];
+      // Trim out lines that reference environment variables that aren't defined
+      for (const line of npmrcFileLines) {
+        const regex: RegExp = /\$\{([^\}]+)\}/g; // This finds environment variable tokens that look like "${VAR_NAME}"
+        const environmentVariables: string[] | null = line.match(regex);
+        let lineShouldBeTrimmed: boolean = false;
+        if (environmentVariables) {
+          for (const token of environmentVariables) {
+            // Remove the leading "${" and the trailing "}" from the token
+            const environmentVariableName: string = token.substring(2, token.length - 1);
+            if (!process.env[environmentVariableName]) {
+              lineShouldBeTrimmed = true;
+              break;
+            }
+          }
+        }
+
+        if (!lineShouldBeTrimmed) {
+          resultLines.push(line);
+        }
+      }
+
+      fs.writeFileSync(packageInstallNpmrcPath, resultLines.join(os.EOL));
+    } catch (e) {
+      throw new Error(`Error reading or writing .npmrc file: ${e}`);
+    }
+  }
+}
+
+/**
+ * Detects if the package in the specified directory is installed
+ */
+function isPackageAlreadyInstalled(packageInstallFolder: string): boolean {
+  try {
+    const flagFilePath: string = path.join(packageInstallFolder, INSTALLED_FLAG_FILENAME);
+    if (!fs.existsSync(flagFilePath)) {
+      return false;
+    }
+
+    const fileContents: string = fs.readFileSync(flagFilePath).toString();
+    return fileContents.trim() === process.version;
+  } catch (e) {
+    return false;
+  }
+}
+
+/**
+ * Removes the installed.flag file and the node_modules folder under the specified folder path.
+ */
+function cleanInstallFolder(rushCommonFolder: string, packageInstallFolder: string): void {
```

Comment:
>cleanInstallFolder [](start = 9, length = 18)

We should definitely delete any shrinkwrap files that were dropped by the last install